src/include/miopen/median/problem_description.hpp

```diff
@@ -0,0 +1,250 @@
+
+struct NetworkConfig;
+
+namespace median {
+
+struct FwdProblemDescription : public ProblemDescriptionBase
+{
+    FwdProblemDescription(const TensorDescriptor& inputDesc_,
+                          const TensorDescriptor& outputDesc_,
+                          const TensorDescriptor& indicesDesc_,
+                          const uint64_t dim_,
+                          const bool keepdim_)
+        : inputDesc(inputDesc_),
+          outputDesc(outputDesc_),
+          indicesDesc(indicesDesc_),
+          dim(dim_),
+          keepdim(keepdim_)
+    {
+        IsValidNumDims();
+        IsValidDim();
+        IsRightLength();
+        IsSameType();
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+    const TensorDescriptor& GetIndicesDesc() const { return indicesDesc; }
+    uint64_t GetDim() const { return dim; }
+    bool GetKeepDim() const { return keepdim; }
+
+    bool IsValidNumDims() const
+    {
+        if(inputDesc.GetNumDims() > 5)
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "MedianForward: A tensor with dimensions greater than 5 is not supported yet.");
+        }
+
+        return true;
+    }
+
+    bool IsValidDim() const
+    {
+
+        if(dim >= inputDesc.GetNumDims())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Invalid dim " + std::to_string(dim) +
+                             " for input tensor with " + std::to_string(inputDesc.GetNumDims()) +
+                             " dimensions");
+        }
+
+        return true;
+    }
+
+    bool IsRightLength() const
+    {
+        auto input_dims   = inputDesc.GetLengths();
+        auto output_dims  = outputDesc.GetLengths();
+        auto indices_dims = indicesDesc.GetLengths();
+
+        auto desired_dims = input_dims;
+        if(keepdim)
+        {
+            desired_dims[dim] = 1;
+        }
+        else
+        {
+            desired_dims.erase(desired_dims.begin() + dim);
+            if(desired_dims.size() == 0)
+                desired_dims.push_back(1);
+        }
+
+        if(output_dims != desired_dims || indices_dims != desired_dims)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Input and output/indices tensor dimension lengths do "
+                         "not match.");
+        }
+
+        return true;
+    }
+
+    bool IsSameType() const
+    {
+        if(inputDesc.GetType() != outputDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Input and output tensor data types do not match.");
+        }
+        return true;
+    }
+
+    bool IsValidFloat() const
+    {
+        return inputDesc.GetType() == miopenFloat || inputDesc.GetType() == miopenHalf ||
+               inputDesc.GetType() == miopenBFloat16;
+    }
```

Comment:
Let's move this implementation to `IsApplicable`. It seem like to be a convension to AMD.

Reply:
`IsValidFloat()` is already in `IsApplicable()`, pls check it again

Reply:
Do you have any official AMD reference regarding the convention for implementing this in `IsApplicable()`? If it's only "seems like", rather than being explicitly stated, imo, it would make more sense to implement it here (together with other condition checks) and call it within `IsApplicable()` instead

Reply:
Check [this](https://github.com/ROCm/MIOpen/pull/3247#discussion_r1765200151)

Reply:
solved